### PR TITLE
Cover DirectoryHelpers with unit tests #16

### PR DESCRIPTION
**Pull request by leonpw**

### CODE REVIEW

File: Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/EmptyIfNullTests.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿namespace MoreDotNet.Tests.Extensions.Collections.EnumerableExtensions
+{
+    using MoreDotNet.Extensions.Collections;
+    using Xunit;
+
+    public class EmptyIfNullTests
+    {
+        [Fact]
+        public void EmtyIfNull_ShouldReturnEqualCollection()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/EmptyIfNullTests.cs]: There's a typo here

File: Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/EmptyIfNullTests.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            int[] input = new[] { 1, 2, 3 };
+            var actual = input.EmptyIfNull();
+            Assert.Equal(input, actual);
+        }
+
+        [Fact]
+        public void EmptyIfNull_ShouldReturnEmptyIfNull()
+        {
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/EmptyIfNullTests.cs:18]: Use the following naming convention for tests:
[MethodName]_[TestSetup]_Should[ExpectedResult]
So basically:
EmtyIfNull_ShouldReturnEqualCollection
can be written as:
 EmtyIfNull_WhenGivenNonNullEnumerable_ShouldReturnEqualCollection

File: Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/ForEachTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            EnumerableExtensions.ForEach(input, (_) => actual.Add(_));
+            Assert.True(actual.Contains(input[0]));
+        }
+
+        [Fact]
+        public void ForEach_ShouldThrowOnNullItems()
+        {
+            Assert.Throws(
+                typeof(ArgumentNullException),
+                () => EnumerableExtensions.ForEach<int>(null, null));
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/ForEachTests.cs:23]: When asserting for exceptions, please use the generic overload:
Assert.Throws<<T>>
Example:
`Assert.Throws(typeof(ArgumentNullException), () => EnumerableExtensions.ForEach<int>(null, null));`
can be written as:
`Assert.Throws<ArgumentNullException>(() => EnumerableExtensions.ForEach<int>(null, null));`

**Review by Teodor92 — CHANGES_REQUESTED** [review]
Looks good - if could do the minor style changes, I'll merge it in :)

P.S. Sorry about the missing contribution/code style guidelines - will add them as soon as possible.

**Comment by leonpw** [comment]
Okay, I fixed it. Thanks for your comments!

**Pull request merged**